.github/workflows/run-eval.yml

```diff
@@ -0,0 +1,39 @@
+# Run evaluation on a PR
+name: Run Eval
+
+# Runs when a PR is labeled with one of the "run-eval-" labels
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  trigger-job:
+    name: Trigger remote eval job
+    # TODO: Add different labels for different eval instances
+    if: ${{ github.event.label.name == 'run-eval-s' }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Trigger remote job
+        run: |
```

Comment:
does the normal `uses:` syntax not work here? e.g. I have this in another repo to trigger actions across repos

```
uses: all-hands-ai/deploy/.github/workflows/_docker_push.yaml@main
```

Reply:
It's cause the remote workflow is not a reusable workflow. It can be used as a standalone as well.

Reply:
Can you please explain a bit more, why would this workflow not work in this repo?

Reply:
I mean that the remote workflow is not of the type reusable workflow so you can't call it like that.
The remote workflow is a workflow that can be triggered manually.

Or did you mean why can we not bring the remote workflow into this repo?

Reply:
I'm totally confused, sorry if I confuse you too. What I mean is just this:

- I assumed this PR is fixing https://github.com/All-Hands-AI/OpenHands/issues/5106 
- this is a new workflow file, but we had [this one](https://github.com/All-Hands-AI/OpenHands/blob/main/.github/workflows/eval-runner.yml)
- I think 5106 has an awesome idea, haiku would be great since deepseek wasn't working eventually!

Maybe I just assumed wrong, and this PR was never intended for that eval? 😅

Reply:
Oh! That eval workflow you mentioned is going to be replaced by this one. 

* The default model is sonnet 3.5 right now but that will likely change to Haiku based on what I've been told.
* This workflow will probably have multiple labels. Depending on label it might run 5, 30, 100, 300 instances.

Will talk to the team on iterating on this. This gets the `run-eval-s` label working and running the eval :) 

Does that make sense?

Reply:
How would Haiku be qualified to run any evaluation if Sonnet is the gold standard? 🤔

Reply:
@tobitege IMHO there are quite a few cases where we need to run a mini-eval, for debugging or as sanity check, e.g. to double check that the behavior didn't change when it shouldn't change.
I don't think it's necessary to run all 300 tasks on swe-bench for example. Depending on the case, I find from 1 to 93 tasks, a run or min-run could be useful.

Haiku 3.5 has function calling, so running with it would show if something broke on the same kind of code/prompts/agent that is normally run in the "real" evals. Moreover, simply comparing previous results would show if there's a suspicious difference.

Reply:
Supporting tools doesn't mean it is _good_ at it, e.g. see [Slack discussion](https://openhands-ai.slack.com/archives/C06PB3T5ZK6/p1731940241098899).
I'm not arguing against running an eval, but against using Haiku as it is not a good model for programming.

Reply:
For now the default is still sonnet-3-5.

Reply:
Tobi, I think you're right as far as Haiku goes. I also think we talk about different things? I'm not saying that Haiku would replace Sonnet 3.5 evals _when Sonnet is needed_. That's not my argument and not what adding Haiku here implies, afaict.

I look at it more as an addition: when we don't need Sonnet but want to see something.

If the examples above are not good, what do you think about another: a nightly run. If Haiku does, let's say, 19% usually, then seeing a nightly with 2% is a signal that something went wrong. That's all. A signal is all we need to double check on a real eval, like on Sonnet - or whatever we have to do.

Haiku doesn't need to be good, to serve as signal this way? It just needs to be better than a coin flip. 🤔 😅

Just my opinion anyway. I will be even happier when we find/decide/start evals on an open LLM.

Reply:
- Who decides and with which criteria, that model a or b is "needed"?
- If cost is an issue, gpt4o is still cheaper than Sonnet, but more capable than Haiku.
- The signal argument I don't understand: what if changes to the codebase (e.g. by a PR) would work fine or better than previous runs in Sonnet, but a smaller model (like Haiku) has more problems with it and causes the result to drop. If this is possible, then the "signal" could lead to wrong conclusions (about the PR)? Tl;dr what if the PR changes have nothing to do with the "bad signal", but just the llm not being good enough?
- IMO using an inferior model only adds potential for extra errors or unpredictable results

Reply:
We definitely need a "cheap" eval option, which can be used as a proxy for evals on Sonnet. We can do a lot more ad hoc testing if we use cheap models, which would be cost-prohibitive with Sonnet.

Of course, this carries the assumption that results will _correlate_ between models. And they mostly do, but ofc we've all seen examples where the correlation breaks down. So we will always need to run a final eval on Sonnet to ensure there aren't any regressions with our default model.